src/ui/suspense/frameCache.ts

```diff
@@ -60,3 +81,99 @@ function createPauseFrames(pauseId: PauseId, frames: Frame[], sourcesState: Sour
     sourcesState.sourceDetails.entities
   )!;
 }
+
+export interface FormattedPointStackFrame {
+  url?: string;
+  source: Source;
+  functionLocation: Location;
+  executionLocation: Location;
+  point: PointDescription;
+}
+
+export interface FormattedPointStack {
+  point: PointDescription;
+  resultPoint: PointDescription;
+  frame?: FormattedPointStackFrame;
+  allFrames: FormattedPointStackFrame[];
+  filteredFrames: FormattedPointStackFrame[];
+  functionName?: string;
+}
+
+export function formatPointStackFrame(
+  frame: PointStackFrame,
+  sourcesById: Map<string, Source>
+): FormattedPointStackFrame {
+  updateMappedLocationForPointStackFrame(sourcesById, frame);
+  const functionLocation = getPreferredLocation(sourcesById, [], frame.functionLocation);
+  const executionLocation = getPreferredLocation(sourcesById, [], frame.point.frame ?? []);
+
+  const source = sourcesById.get(functionLocation.sourceId)!;
+
+  return {
+    url: source.url,
+    source,
+    functionLocation,
+    executionLocation,
+    point: frame.point,
+  };
+}
+
+export const formattedPointStackCache: Cache<
+  [replayClient: ReplayClientInterface, point: TimeStampedPoint],
+  FormattedPointStack
+> = createCache({
+  debugLabel: "FormattedPointStack",
+  getKey: ([replayClient, point]) => point.point,
+  async load([replayClient, point]) {
+    return formatPointStackForPoint(replayClient, point);
+  },
+});
```

Comment:
Nothing referencing this cache?

```suggestion
```

Reply:
I added the cache after writing the standalone function.  The other logic should really be using the cache for consistency - I'll update that.